### PR TITLE
25-test_x509.t: Re-add and improve a test on non-existence of ASN.1parse errors

### PR DESCRIPTION
**Pull request by DDvO**
This corrects the 'fix' done in #13309 and makes more clear what the test is about and how it works.

### CODE REVIEW

File: test/recipes/25-test_x509.t [hunk 1]
```diff
@@ -124,8 +124,12 @@ sub test_errors { # actually tests diagnostics of OSSL_STORE
     return $res && $found;
 }
 
+# 3 tests for non-existence of spurious OSSL_STORE ASN.1 parse error output.
+# This requires provoking somehow a non-successful late exit of the app.
```
Comment (paulidale) [inline test/recipes/25-test_x509.t:128]: I'd drop the _somehow_ here.

**Review by paulidale — APPROVED** [review]
One wording suggestion in a comment but otherwise looks good.

**Review by paulidale — APPROVED** [review]
Even better.

File: test/recipes/25-test_x509.t (hunk 1 above)
Comment (DDvO) [reply test/recipes/25-test_x509.t]: I've just pushed a fixup that drops "somehow" and adds a little more detail:
```
# This requires provoking a failure exit of the app after reading input files.
```

**Comment by DDvO** [comment]
Thanks @paulidale for your immediate review and approval!

**Comment by DDvO** [comment]
Merged - thanks

**Pull request closed without merging**